src/components/Identity/configurePayload.js

```diff
@@ -0,0 +1,39 @@
+import getBrowser from "../../utils/getBrowser";
+import areThirdPartyCookiesSupportedByDefault from "../../utils/areThirdPartyCookiesSupportedByDefault";
+import ecidNamespace from "../../constants/ecidNamespace";
+
+const addEcidToPayload = (payload, ecid) => {
+  payload.addIdentity(ecidNamespace, {
+    id: ecid
+  });
+};
+export default (
+  payload,
+  migration,
+  idMigrationEnabled,
+  thirdPartyCookiesEnabled
+) => {
+  const ecidToMigrate =
+    idMigrationEnabled && migration.getEcidFromLegacyCookies();
+  if (ecidToMigrate) {
+    // We have an ECID, but we still want to establish an
+    // identity cookie before allowing other requests to be sent.
+    addEcidToPayload(payload, ecidToMigrate);
+  }
```

Comment:
This `configurePayload` module is used for the `getEcid` payload (as well as other payloads), but if we already have an ecid to migrate (`ecidToMigrate` exists), then there's no need to issue a getEcid request in the first place.